src/ClientSideMonitoring/ConfigurationProvider.php

```diff
@@ -77,11 +77,15 @@ class ConfigurationProvider extends AbstractConfigurationProvider
      */
     public static function defaultProvider(array $config = [])
     {
-        $configProviders = [
-            self::env(),
-            self::ini(),
-            self::fallback()
-        ];
+        $configProviders = [];
+        array_push($configProviders, self::env());
+        if (
+            !isset($config['use_aws_config_file'])
+            || $config['use_aws_config_file'] != false
+        ) {
+            array_push($configProviders, self::ini());
```

Comment:
Since they're effectively the same, I checked and found one usage of array_push in the src folder and modified that as well; it was in src/S3/PostObjectV4.  I figured since there was only a minor change it wouldn't need its own PR, but let me know if you want me to revert that or anything